dlib/dnn/input_abstract.h

```diff
@@ -271,6 +271,155 @@ namespace dlib
 
     };
 
+// ----------------------------------------------------------------------------------------
+
+    template <
+        typename PYRAMID_TYPE
+        >
+    class input_grayscale_image_pyramid
+    {
+        /*!
+            REQUIREMENTS ON PYRAMID_TYPE
+                PYRAMID_TYPE must be an instance of the dlib::pyramid_down template.
+
+            WHAT THIS OBJECT REPRESENTS
+                This input layer works with gray scale images of type matrix<unsigned char>.
+                It is identical to input except that it outputs a tensor containing a tiled
```

Comment:
Say "identical to the input layer except..."

Reply:
Sorry, but I don't understand what you mean.

Reply:
I just mean to replace "It is identical to input except that it outputs a tensor" with "It is identical to the input layer except that it outputs a tensor"